packages/synapse-react-client/src/utils/hooks/useGetGoalData.test.ts

```diff
@@ -0,0 +1,143 @@
+        { id: '123113', name: 'Summary', columnType: 'LARGETEXT' },
+        { id: '123022', name: 'TableId', columnType: 'ENTITYID' },
+        { id: '123023', name: 'Asset', columnType: 'FILEHANDLEID' },
+        { id: '123021', name: 'Link', columnType: 'STRING' },
+        { id: '123043', name: 'ItemOrder', columnType: 'STRING' },
+      ],
+      rows: [
+        {
+          rowId: 1,
+          versionNumber: 3,
+          values: [
+            'Methods',
+            'Question-driven studies that define the scientific scope and provide context to the contributed resources.',
+            'syn21783965',
+            '64155356',
+            'Explore/Studies',
+            '1',
+          ],
+        },
+        {
+          rowId: 2,
+          versionNumber: 3,
+          values: [
+            'Datasets & Files',
+            'Collected from samples and cell lines across a spectrum of genomic assays and neuropsychiatric diseases.',
+            'syn20821313',
+            '64155359',
+            'Explore/Data',
+            '2',
+          ],
+        },
+        {
+          rowId: 3,
+          versionNumber: 3,
+          values: [
+            'Publications',
+            'Lessons learned from consortia data and methods, shared as peer-reviewed journal articles.',
+            'syn22095937',
+            '64155361',
+            'Explore/Publications',
+            '3',
+          ],
+        },
+        {
+          rowId: 4,
+          versionNumber: 4,
+          values: [
+            'People',
+            'Funded grant programs comprising institutions and investigators that are pursuing neuropsychiatric disease research themes with multidimensional approaches.',
+            'syn22096112',
+            '64155368',
+            'Explore/People',
+            '4',
+          ],
+        },
+      ],
+    },
+  },
+}
+
+const mockFileResult = [
+  {
+    fileHandleId: '149976034',
+    preSignedURL: 'https://mockurl.com/orangecat.jpeg',
+  },
+  {
+    fileHandleId: '149976044',
+    preSignedURL: 'https://mockurl.com/tabbycat.jpeg',
+  },
+]
+
+const mockBatchFileResult: BatchFileResult = {
+  requestedFiles: mockFileResult,
+}
+
+describe('useGetGoalData', () => {
+  beforeAll(() => {
+    server.listen()
+  })
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest.spyOn(SynapseClient, 'getFiles').mockResolvedValue(mockBatchFileResult)
+  })
+  it('should return goal data', async () => {
+    const entityId = 'syn22315959'
+    let result: {
+      current: any
+      all?: (
+        | Error
+        | {
+            assets: string[] | undefined
+            error: string | SynapseClientError | undefined
+          }
+      )[]
+      error?: Error | undefined
+    }
+    await waitFor(() => {
+      result = renderHook(() => useGetGoalData(entityId, tableQueryResult), {
+        wrapper: createWrapper(),
+      }).result
+    })
```

Comment:
Can you just do this? This `waitFor` isn't really waiting for anything--if it's just to get rid of the `act` warning, we need a different waitFor called _after_ `renderHook`
```suggestion
    const { result } = renderHook(() => useGetGoalData(entityId, tableQueryResult), {
        wrapper: createWrapper(),
    })
```